src/client/dfuse/pil4dfs/int_dfs.c

```diff
@@ -6355,6 +6976,42 @@ register_handler(int sig, struct sigaction *old_handler)
 	}
 }
 
+/* Check whether current executable is sh/bash or not. Flag is_bash is set. */
+static void
+check_exe_sh_bash(void)
+{
+	int   readsize;
+	/* the exe name extract from /proc/self/exe */
+	char *exe_path;
+	/* the short exe name from exe_path */
+	char *exe_short = NULL;
+
+	D_ALLOC(exe_path, DFS_MAX_PATH);
+	if (exe_path == NULL)
+		return;
+
+	readsize = readlink("/proc/self/exe", exe_path, DFS_MAX_PATH);
+	if (readsize >= DFS_MAX_PATH) {
+		DS_ERROR(ENAMETOOLONG, "path from readlink() is too long");
+		goto out;
+	} else if (readsize < 0) {
+		DS_ERROR(errno, "readlink() failed");
+		goto out;
+	}
+
+	exe_short = basename(exe_path);
+	if (exe_short == NULL) {
+		DS_ERROR(errno, "basename() failed");
+		goto out;
+	}
+	if (strncmp(exe_short, "bash", 5) == 0 || strncmp(exe_short, "sh", 3) == 0)
```

Comment:
Could you use `STR_AND_SIZE` here?

Reply:
These two strings are very short, so I prefer not using STR_AND_SIZE.